client-oauth2.js

```diff
@@ -703,6 +711,59 @@
   };
 
   /**
+   * Support JSON Web Token (JWT) Bearer Token OAuth 2.0 grant.
+   *
+   * Reference: https://tools.ietf.org/html/draft-ietf-oauth-jwt-bearer-12#section-2.1
+   *
+   * @param {ClientOAuth2} client
+   */
+  function JwtBearerFlow (client) {
+    this.client = client;
+  }
+
+  /**
+   * Request an access token using a JWT token.
+   *
+   * @param  {string} token A JWT token.
+   * @param  {Object}  [options]
+   * @return {Promise}
+   */
+  JwtBearerFlow.prototype.getToken = function (token, options) {
+    var self = this;
+
+    options = assign({}, this.client.options, options);
+
+    expects(options, [
+      'accessTokenUri'
+    ]);
+
+    var headers = {
+      'Accept':        'application/json, application/x-www-form-urlencoded',
+      'Content-Type':  'application/x-www-form-urlencoded'
+    };
+
+    // Authentication of the client is optional, as described in Section 3.2.1 of OAuth 2.0 [RFC6749]
+    if (options.clientId) {
+      headers['Authorization'] = 'Basic ' + btoa(options.clientId + ':' + options.clientSecret || '');
```

Comment:
I'm pretty sure you're correct on that. Maybe we should fix all of these later so they won't ever say "undefined"? It's out of scope though, so this fix will be cool :+1: